### ISSUE TITLE
[Bug]: snowflake_account resource error due to 2024_06 bundle change

### ISSUE THREAD

**Issue by sbrudz** [Snowflake-Labs/terraform-provider-snowflake#3016]
### Terraform CLI Version

1.8.5

### Terraform Provider Version

0.94.1

### Terraform Configuration

```terraform
terraform {
  required_version = "1.8.5"
}

terraform {
  required_providers {
    snowflake = {
      source  = "Snowflake-Labs/snowflake"
      version = "0.94.1"
    }
  }
}

variable "snowflake_org_user" {
  type = string
}

variable "snowflake_org_account" {
  type = string
}

variable "admin_password" {
  type      = string
  sensitive = true
}

provider "snowflake" {
  user          = var.snowflake_org_user
  account       = var.snowflake_org_account
  role          = "ORGADMIN"
}

resource "snowflake_account" "this" {
  name           = "BUG_TEST"
  admin_name     = "TEST_ADMIN"
  admin_password = var.admin_password
  email          = "test@xample.com"
  comment        = "Managed by Terraform"
  edition        = "STANDARD"
}
```

### Category

category:sdk

### Object type(s)

resource:account

### Expected Behavior

Running `terraform apply` on the above configuration and confirming the plan should create a new account and complete without error.

### Actual Behavior

As of 8/27, when I run `terraform apply` and confirm the plan, I get the following error:
```
Error: giving up after 5 attempts
 
  with snowflake_account.this,
  on main.tf line 33, in resource "snowflake_account" "this":
  33: resource "snowflake_account" "this" {
```
The account does get created, but the state isn't saved because of the error.

### Steps to Reproduce

To reproduce the error:
1. Run `SELECT SYSTEM$SHOW_ACTIVE_BEHAVIOR_CHANGE_BUNDLES();` to check that the `2024_06` bundle is enabled in your account.  If not, run `SELECT SYSTEM$ENABLE_BEHAVIOR_CHANGE_BUNDLE('2024_06');`
2. Run the provided code as a user with the org_admin role
3. Run `terraform apply` and confirm the plan

### How much impact is this issue causing?

Medium

### Logs

https://gist.github.com/sbrudz/363aedc61c209e86958e67d2aad3350a

### Additional Information

The 2024_06 bundle became enabled by default on August 26-27.  This bundle [changes the behavior](https://docs.snowflake.com/en/release-notes/bcr-bundles/2024_06/bcr-1712) of the `SHOW ORGANIZATION ACCOUNTS` command.

The workaround is to disable the 2024_06 bundle:
```
SELECT SYSTEM$DISABLE_BEHAVIOR_CHANGE_BUNDLE('2024_06');
```

It seems like this could be fixed by using `SHOW ACCOUNTS` instead of `SHOW ORGANIZATION ACCOUNTS` within the [Accounts SDK](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/v0.94.1/pkg/sdk/accounts.go#L265).

### Would you like to implement a fix?

- [ ] Yeah, I'll take it 😎

**Comment by sfc-gh-jcieslak** [comment]
Hey @sbrudz 👋 
Thank you for reporting this issue. I'll change the SHOW command to the newer one. The fix should be present in the newest release.

**Comment by sbrudz** [comment]
Excellent! Thank you, @sfc-gh-jcieslak!

**Comment by marcofunaro** [comment]
@sfc-gh-jcieslak thanks for the prompt response! 

As far as you know, is there an official policy to align the release of this provider with the official Snowflake releases? I'm creating an internal process around Snowflake updates.

**Comment by sfc-gh-asawicki** [comment]
Hey @marcofunaro. You are right, there is no alignment currently. We will discuss the release's cadence after reaching a stable V1.

**Comment by marcofunaro** [comment]
do you want me to open an issue for tracking purposes?

**Comment by asafcyclops** [comment]
As a side note, we have also encountered this issue and it was very hard to debug.
It appears that the current implementation is swallowing the error and is not logging it in any way making it hard do understand the root cause. would be nice if we could have an error message here when retrying.

**Comment by sfc-gh-jcieslak** [comment]
Hey all 👋
Yesterday, we released a new provider version ([v0.95.0](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.95.0)). Please upgrade and let us know if the issue disappeared. 

@marcofunaro I think there's no need for that. We are aware of the issue and will communicate to you any decision we'll make on that most likely through GitHub discussions. 

@asafcyclops Thank You for pointing this out, we already created a task for that and will try to address it as part of the next release.

**Comment by sbrudz** [comment]
Confirming that the issue is fixed for us after upgrading to v0.95.0.  Thank you, @sfc-gh-jcieslak!
